lib/call-cabal-project-to-nix.nix

```diff
@@ -391,10 +412,11 @@ let
     export GIT_SSL_CAINFO=${cacert}/etc/ssl/certs/ca-bundle.crt
     HOME=${dotCabal {
       inherit cabal-install nix-tools extra-hackage-tarballs;
-      index-state =
-        builtins.trace ("Using index-state: ${index-state-found}" + (if name == null then "" else " for " + name))
-          index-state-found;
+      index-state = cached-index-state;
       sha256 = index-sha256-found; }} cabal v2-configure \
+        --index-state=${
+            builtins.trace ("Using index-state: ${index-state-found}" + (if name == null then "" else " for " + name))
+              index-state-found} \
```

Comment:
Does this definitely work, even if the `index-state` in `dotCabal` is different?

Generally this is quite subtle, and it's pretty non-obvious why we're picking one over the other. Could we get a longer comment explaining the choices?

Reply:
I have added comments describing how index-state in dotCabal and the one passed to cabal v2-configure interact.